tests/NgTests/StorageFactoryTests.cs

```diff
@@ -0,0 +1,115 @@
+        public void AzureCommpressedFactoryNull(string storageBaseAddress,
+                                            string storageAccountName,
+                                            string storageKeyValue,
+                                            string storageContainer,
+                                            string storagePath,
+                                            string storageType)
+        {
+            Dictionary<string, string> arguments = new Dictionary<string, string>()
+            {
+                { Arguments.CompressedStorageBaseAddress, storageBaseAddress },
+                { Arguments.CompressedStorageAccountName, storageAccountName },
+                { Arguments.CompressedStorageKeyValue, storageKeyValue },
+                { Arguments.CompressedStorageContainer, storageContainer },
+                { Arguments.CompressedStoragePath, storagePath },
+                { Arguments.StorageType, storageType},
+                { Arguments.UseCompressedStorage, "false"}
+            };
+
+            StorageFactory factory = CommandHelpers.CreateCompressedStorageFactory(arguments, true);
+            // Assert
+            Assert.True(factory == null, "The CompressedStorageFactory should be null when the UseCompressedStorage is false.");
+        }
+
+        [Theory]
+        [Description("The regular azure factory should not compress the content if.")]
+        [InlineData("http://localhost/reg", "testAccount", "ZiWrbTMNTApZxLeWB7a0yN52gJj+ZlGE0ipRi9PaTcn9AU4epwvsngE5rLSMk9TwpazxUtzeyBnFeWFAhfkGpw==", "testContainer", "testStoragePath", "azure")]
+        public void AzureFactory(string storageBaseAddress,
+                                 string storageAccountName,
+                                 string storageKeyValue,
+                                 string storageContainer,
+                                 string storagePath,
+                                 string storageType)
+        {
+            Dictionary<string, string> arguments = new Dictionary<string, string>()
+            {
+                { Arguments.StorageBaseAddress, storageBaseAddress },
+                { Arguments.StorageAccountName, storageAccountName },
+                { Arguments.StorageKeyValue, storageKeyValue },
+                { Arguments.StorageContainer, storageContainer },
+                { Arguments.StoragePath, storagePath },
+                { Arguments.StorageType, storageType}
+            };
+
+            StorageFactory factory = CommandHelpers.CreateStorageFactory(arguments, true);
+            AzureStorageFactory azureFactory = factory as AzureStorageFactory;
+            // Assert
+            Assert.True(azureFactory != null);
+            Assert.False(azureFactory.CompressContent, "The azure storage factory should not compress the content.");
+        }
+    }
+
+    public class Arguments
+    {
```

Comment:
There is a private version in ng. I preferred to add what needed at the test level instead of making that public.